kms/client.go

```diff
@@ -722,6 +723,67 @@ func (c *Client) ReadDB(ctx context.Context) (*ReadDBResponse, error) {
 	}, nil
 }
 
+// Logs returns a stream of server log records from req.Host.
+// If req.Host is empty, the first host of the client's host
+// list is used.
+//
+// The LogRequest specifies which log records are fetched.
+// For example, only records with a certain log level or
+// records that contain a specific log message.
+//
+// It requires SysAdmin privileges.
+//
+// The returned error is of type *HostError.
+func (c *Client) Logs(ctx context.Context, req *LogRequest) (*LogResponse, error) {
+	const (
+		Method   = http.MethodPost
+		Path     = api.PathLog
+		StatusOK = http.StatusOK
+	)
+
+	var (
+		err    error
+		reqURL string
+		host   = req.Host
+	)
+	if host == "" {
+		reqURL, host, err = c.lb.URL(Path)
+	} else {
+		reqURL, err = url.JoinPath(httpsURL(host), Path)
+	}
+	if err != nil {
+		return nil, hostError(host, err)
+	}
+
+	body, err := pb.Marshal(req)
+	if err != nil {
+		return nil, hostError(host, err)
+	}
```

Comment:
What's the rationale behind having a REST API that uses protobuf for the request body? It looks like the response is a custom binary format. Why not send JSON and return JSON (or ND-JSON) instead?

Reply:
The KMS can / will be able to do this. It supports protobuf / nd-json and plain text formats.
Protobuf is more compact, more efficient and comes for free since the JSON representation is also done via protobuf. (You don't need gRPC to use an efficient binary format 😉 ). The SDK simply doesn't use JSON because it can rely on the fact that the server supports protobuf for all APIs.

Reply:
Also, the lenght-prefixed encoding is a commen tech. to implement protobuf streams.

Reply:
If the underlying REST API can do NDJSON too, then it's fine. I can imagine you'll use the more efficient method in the Go API.